File: 프로그래머스/unrated/181856. 배열 비교하기/배열 비교하기.py
def solution(arr1, arr2):
    if len(arr1) > len(arr2):
        return 1
    elif len(arr1) < len(arr2):
        return -1
    elif sum(arr1) > sum(arr2):
        return 1
    elif sum(arr1) < sum(arr2):
        return -1
    else:
        return 0
    
    # # 방법 1 - 배열 합을 변수에 저장
    # else:
    #     sum1 = sum(arr1)
    #     sum2 = sum(arr2)
    #     if sum1 > sum2:
    #         return 1
    #     elif sum1 < sum2:
    #         return -1
    #     else:
    #         return 0

File: 프로그래머스/unrated/181856. 배열 비교하기/배열 비교하기.js
function solution(arr1, arr2) {
    if (arr1.length > arr2.length) {
        return 1
    } else if (arr1.length < arr2.length) {
        return -1
    } else {
        const sum1 = arr1.reduce((acc, el) => acc += el, 0)
        const sum2 = arr2.reduce((acc, el) => acc += el, 0)
        
        return sum1 > sum2 ? 1 : sum1 < sum2 ? -1 : 0
    }
}